app/admin-tab/machine/controller.js

```diff
@@ -0,0 +1,128 @@
+import Ember from 'ember';
+import Sortable from 'ui/mixins/sortable';
+import C from 'ui/utils/constants';
+
+export default Ember.Controller.extend(Sortable, {
+  application: Ember.inject.controller(),
+  settings:    Ember.inject.service(),
+  newMachine: null,
+  sortBy: 'name',
+  sorts: {name: ['name']},
+  upgrading: false,
+
+  actions: {
+    activate: function(driver) {
+      let action = null;
+      if (driver.get('actionLinks.activate')) {
+        action = 'activate';
+      } else if (driver.get('actionLinks.reactivate')) {
+        action = 'reactivate';
+      }
+
+      driver.doAction(action);
+    },
+    addNewDriver: function(driver) {
+      let newDriver = {
```

Comment:
nit: I would normally make `newDriver` the store record instead of a pojo so you can't accidentally use the wrong one further down